runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -2043,7 +2043,7 @@ TR_OpaqueClassBlock *
 TR_J9SharedCacheServerVM::getSystemClassFromClassName(const char * name, int32_t length, bool isVettedForAOT)
    {
    TR::Compilation *comp = _compInfoPT->getCompilation();
-   TR_OpaqueClassBlock *classPointer = TR_J9ServerVM::getSystemClassFromClassName(name, length);
+   TR_OpaqueClassBlock *classPointer = TR_J9ServerVM::getSystemClassFromClassName(name, length, isVettedForAOT);
```

Comment:
Looking at the code I see that in the `TR_J9SharedCacheVM` implementation of `getSystemClassFromClassName()` we call `TR_J9VM::getSystemClassFromClassName()` with the default value of the `isVettedForAOT` which is `false`, but actually it doesn't matter because `TR_J9VM::getSystemClassFromClassName()` doesn't even look at `isVettedForAOT`.
In contrast the TR_J9Server implementation sends a message to the client with `isVettedForAOT` and the client executes the corresponding query using an AOT frontend for which `isVettedForAOT` may matter.
So the problem is that JITServer executes a `TR_J9VM` query which is replied by the client with an answer from a `TR_J9SharedCacheVM` frontend. Since the `MessageType::VM_getSystemClassFromClassName` is only sent by the JITServer from `TR_J9ServerVM::getSystemClassFromClassName` (i.e. the non-AOT frontend) I think the right solution is for the client to create a non-AOT frontend (TR_J9VM) to find the answer to the query.

Reply:
Changed this to call non-AOT front end now: 5608aa7